src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,193 @@
+import { BadgeProps, BadgeTypeMap } from './Badge.types';
+
+import clsx from 'clsx';
+import { badgeClasses, getBadgeUtilityClass } from './Badge.classes';
+
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+
+import { useThemeProps } from '@mui/material/styles';
+import { keyframes } from '@mui/system';
+import { ButtonBase, capitalize, styled, touchRippleClasses } from '@mui/material';
+import { OverridableComponent } from '@mui/material/OverridableComponent';
+
+const enterKeyframe = keyframes`
+  0% {
+    transform: scale(0);
+    opacity: 0.1;
+  }
+  100% {
+    transform: scale(1);
+    opacity: 1;
+  }
+`;
+
+type BadgeOwnerState = {
+  classes?: BadgeProps['classes'];
+  color: NonNullable<BadgeProps['color']>;
+  size: NonNullable<BadgeProps['size']>;
+};
+
+const useUtilityClasses = (ownerState: BadgeOwnerState) => {
+  const { classes, color, size } = ownerState;
+
+  const slots = {
+    root: ['root', `color${capitalize(color)}`, `size${size}`]
+  };
+
+  return composeClasses(slots, getBadgeUtilityClass, classes);
+};
+
+const BadgeRoot = styled(ButtonBase, {
+  name: 'ESBadge',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const {
+      ownerState: { color, variant, severity }
+    } = props;
+
+    return [styles.root, styles[variant], styles[`${variant}${capitalize(color || severity)}`]];
```

Comment:
Копипастнул, но под наш компонент вообще не подстроил =)
```suggestion
    return [styles.root, styles[`colors${capitalize(color)}`], styles[`size${size}`]];
```